### DISCUSSION TITLE
strptime with timezone

### DISCUSSION THREAD

**Discussion by kiliconsult** [duckdb/duckdb#3817] [Q&A]
It seems that strptime ignores timezones.

If i do 
SELECT date_part('epoch', strptime('2022-06-06 CEST', '%Y-%m-%d %Z'))*1000
or
SELECT date_part('epoch', strptime('2022-06-06 GMT+2', '%Y-%m-%d %Z'))*1000
or 
SELECT date_part('epoch', strptime('2022-06-06 Europe/Copenhagen', '%Y-%m-%d %Z'))*1000

All of these are parsed to a UTC date and ignores the timezone.

GMT: Monday 6. June 2022 00:00:00
Your time zone: mandag d. 6. juni 2022 kl. 02:00:00 [GMT+02:00](https://www.epochconverter.com/timezones?q=1654473600000) DST

Does anyone know if I am doing anything wrong or if this is a bug?

**Comment by orzom411** [comment]
Sorry I don't have a suggestion for mitigating the time zone name issue, I'll add I found similar but fortunately we've made everything UTC so it's a matter of parsing the data to provide just the offsets.  Caution, zero padding is required, e.g.:

SELECT date_part('epoch', strptime('2022-06-06 +2', '%Y-%m-%d %z')) as result;

Error: Invalid Input Error: Could not parse string "2022-06-06 +2" according to format specifier "%Y-%m-%d %z"
2022-06-06 +2

This does work:
SELECT date_part('epoch', strptime('2022-06-06 +02', '%Y-%m-%d %z')) as result;

result
1654466400

Results based on library version: v0.3.4-dev2
